### PR TITLE
Update for OJS 3.4

### PR DESCRIPTION
**Pull request by v0k1c**

### CODE REVIEW

File: WOSHandler.php [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+            // Get the comments associated with this review assignment
+            $submissionComments = $submissionCommentDao->getSubmissionComments($submissionId, SubmissionComment::COMMENT_TYPE_PEER_REVIEW, $reviewId);
+            if ($submissionComments) {
+                foreach ($submissionComments->toArray() as $comment) {
+                    // If the comment is viewable by the author, then add the comment.
+                    if ($comment->getViewable()) $body .= PKPString::html2text($comment->getComments()) . "\n";
+                }
+            }
+            if ($reviewFormId = $reviewAssignment->getReviewFormId()) {
+//                $reviewId = $reviewAssignment->getId();
```
Comment (Alex-nz) [inline WOSHandler.php]:
```suggestion
```
Should this be removed?

File: WOSHandler.php [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+            $auth_key = $plugin->getSetting($journalId, 'auth_key');
+            $auth_token = $plugin->getSetting($journalId, 'auth_token');
+
+            $body = str_replace("\r", '', $body);
+            $body = str_replace("\n", '\r\n', $body);
+
+            $dateRequested = new DateTime($reviewAssignment->getDateNotified());
+            $dateCompleted = new DateTime($reviewAssignment->getDateCompleted());
+
+//            $locale = AppLocale::getLocale();
```
Comment (Alex-nz) [inline WOSHandler.php]:
```suggestion
```
And this?

File: WebOfSciencePlugin.php [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+            parent::getActions($request, $actionArgs)
+        );
+    }
+
+    /**
+     * @see GenericPlugin::manage()
+     */
+    function manage($args, $request) {
+//        $templateManager = TemplateManager::getManager($request);
+//        $templateManager->registerPlugin('function', 'plugin_url', [$this, 'smartyPluginUrl']);
```
Comment (Alex-nz) [inline WebOfSciencePlugin.php]:
```suggestion
```
And this?

File: classes/WOSReview.php [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+		return $this->getData('submissionId');
+	}
+
+	/**
+	 * Set the submission ID of the review
+     *
+	 * @param $submissionId int
+	 */
+	function setSubmissionId(int $submissionId): void
+    {
```
Comment (Alex-nz) [inline classes/WOSReview.php:38]: I don't know PHP, but I assume indentation doesn't matter much? It's a little bit off in some functions in this file.

File: gallery.xml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  <maintainer>
+    <name>Clarivate</name>
+    <institution>Clarivate</institution>
+    <email>reviewservices@clarivate.com</email>
+  </maintainer>
+
+  <release date="2024-08-05" version="1.3.4.0" md5="71d1dbed08050f4eb589927f02d8e4ab">
+
+    <package>
+      https://github.com/publons/ojs_3_plugin/releases/download/2021.5.25.0/publons.tar.gz
```
Comment (Alex-nz) [inline gallery.xml:26]: I assume this will be updated once we do the release?

File: locale/pt_BR/locale.po [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+msgstr "Por favor, insira o API key correto para sua integração com Web of Science Reviewer Recognition Service."
+
+msgid "plugins.generic.wosrrs.settings.auth_tokenDescription"
+msgstr "The Authorization token for the specified account (you can find that on https://publons.com/api/v2/ under Token)."
+
+msgid "plugins.generic.wosrrs.settings.authTokenRequired"
+msgstr "Please ensure that your Authorization token is correct."
+
+msgid "plugins.generic.wosrrs.settings.journalToken"
+msgstr "API Key"
```
Comment (Alex-nz) [inline locale/pt_BR/locale.po]:
```suggestion
msgstr "Journal token"
```

**Review by Alex-nz — APPROVED** [review]
Since I am not familiar with PHP, I've left mostly cosmetic comments on the code. Should be good to go for the release

File: classes/WOSReview.php (hunk 4 above)
Comment (v0k1c) [reply classes/WOSReview.php:38]: Yep, it doesn't matter, but I've reformatted the whole file. Probably got some indentations left that I forgot to remove.

File: gallery.xml (hunk 5 above)
Comment (v0k1c) [reply gallery.xml:26]: Yep, date and md5 will need to be updated.

**Pull request merged**